v2/api/network/customizations/load_balancer_extension.go

```diff
@@ -157,17 +254,8 @@ func fuzzySetInboundNatRule(inboundNatRule genruntime.ARMResourceSpec, embeddedI
 		return errors.Wrapf(err, "failed to unmarshal inboundNatRule JSON")
 	}
 
-	// Safety check that these are actually the same:
-	// We can't use reflect.DeepEqual because the types are not the same.
-	embeddedInboundNatRuleJSON, err := json.Marshal(embeddedInboundNatRule.Interface())
-	if err != nil {
-		return errors.Wrap(err, "unable to check that embedded inboundNatRule is the same as inboundNatRule")
-	}
-
-	err = fuzzyEqualityComparison(embeddedInboundNatRuleJSON, inboundNatRuleJSON)
-	if err != nil {
-		return errors.Wrap(err, "failed during comparison for embeddedInboundNatRuleJSON and inboundNatRuleJSON")
-	}
+	// TODO: Can't do a trivial fuzzyEqualityComparison here because we don't know which fields are readonly
```

Comment:
This is my least favorite part of this change. I _think_ it's pretty safe though because before we were using the child resource shape and the parent resource shape and the assertion here was effectively "are they actually the same".

_Now_ we're using the parent resource shape (from GET) and putting that into the parent resource shape (for PUT). Since it's the same resource w/ the same API version, the properties are guaranteed to line-up structurally. Properties that exist on GET but not on PUT are read-only, so it's OK for them to be omitted.